src/plugins/intel_cpu/src/nodes/dft.cpp

```diff
@@ -289,139 +312,278 @@ void DFT::dftNd(float* output, const std::vector<size_t>& outputStrides) const {
             size_t parallelDimIndex = lastDimIndex == currentAxis ? lastDimIndex - 1 : lastDimIndex;
             do {
                 parallel_for(iterationRange[parallelDimIndex], [&](size_t dim) {
-                    std::vector<float> gatheredData(outputLen);
+                    std::vector<float> gatheredData(outputLen * 2);
                     auto parallelIterationCounter = iterationCounter;
                     parallelIterationCounter[parallelDimIndex] = dim;
                     gatherToBufferND(gatheredData.data(), output, currentAxis, parallelIterationCounter, outputShape, outputStrides);
-                    fft(gatheredData.data(), outputLen);
-                    applyBufferND(gatheredData.data(), output, currentAxis, parallelIterationCounter, outputShape, outputStrides);
+                    const auto* bufferPtr =
+                        fft(gatheredData.data(), gatheredData.data() + outputLen, outputLen, inverse);
+                    applyBufferND(bufferPtr, output, currentAxis, parallelIterationCounter, outputShape, outputStrides);
                 });
                 iterationCounter[parallelDimIndex] = iterationRange[parallelDimIndex] - 1;
             } while (nextIterationStep(iterationCounter, iterationRange, currentAxis));
         } else {
             std::vector<float> gatheredData(outputLen);
             do {
                 gatherToBufferND(gatheredData.data(), output, currentAxis, iterationCounter, outputShape, outputStrides);
-                naiveDFT(gatheredData.data(), outputLen);
+                naiveDFT(gatheredData.data(), outputLen, inverse);
                 applyBufferND(gatheredData.data(), output, currentAxis, iterationCounter, outputShape, outputStrides);
             } while (nextIterationStep(iterationCounter, iterationRange, currentAxis));
         }
     }
 }
 
 /* Cooley Tukey implementation of FFT */
-void DFT::fft(float* data, int64_t dataLength, bool parallelize) const {
-    static int cacheSizeL3 = utils::get_cache_size(3, false);
+float* DFT::fft(float* inBuffer,
```

Comment:
NIT. M.b. I'm old-fashioned, but usually when a function returns a raw  non const pointer, that means the caller takes responsibility for the memory and should release the memory itself. M.b. it is better to add yet another parameter with resultBuf ptr, or just state that inBuffer will contain the result.

Reply:
I done it via separate resultBuf pointer